addons/account/models/account_account.py

```diff
@@ -999,26 +1014,29 @@ def _adapt_accounts_for_account_groups(self, account_ids=None, company=None):
         if account_ids:
             account_where_clause = SQL('%s AND account.id IN %s', account_where_clause, tuple(account_ids))
 
-        self._cr.execute(SQL("""
-            WITH relation AS (
-                 SELECT DISTINCT ON (account.id)
-                        account.id AS account_id,
-                        agroup.id AS group_id
-                   FROM account_account account
-                   JOIN res_company account_company ON account_company.id = account.company_id
-              LEFT JOIN account_group agroup
-                     ON agroup.code_prefix_start <= LEFT(account.code, char_length(agroup.code_prefix_start))
-                    AND agroup.code_prefix_end >= LEFT(account.code, char_length(agroup.code_prefix_end))
-                    AND agroup.company_id = split_part(account_company.parent_path, '/', 1)::int
-                  WHERE %s
-               ORDER BY account.id, char_length(agroup.code_prefix_start) DESC, agroup.id
-            )
-            UPDATE account_account
-               SET group_id = rel.group_id
-              FROM relation rel
-             WHERE account_account.id = rel.account_id
-        """, account_where_clause))
-        self.env['account.account'].invalidate_model(['group_id'], flush=False)
+        Account = self.env['account.account']
+        with Account.modifying_model(inputs=['code'], outputs=['group_id']) as m:
```

Comment:
Why don't we flush `account.group` here?
Why are we mixing 2 API?

Reply:
> Because it's already flushed on line 1000.

Then why don't you invalidate under line 1000?

> We're not? The update is on `account_account`, so that's what we're `modifying`.

Then why don't you only call `modified`?

Reply:
> Because that's not where the query is executed?

But that's where the flush is done


> It's what `modifying_model` does?

But it's also what `modified` does
https://github.com/odoo/odoo/pull/155205#discussion_r1706133223